### PR TITLE
Mail contacts: Update public via user contact / display owner's posts

### PR DESCRIPTION
**Pull request by annando**
This touches two issues concerning mail contacts.

1. Updating of public mail accounts can't be done via probing. So we now just take the data from a user contact entry.
2. Displaying of contact posts has to be done via the owner for Discourse posts.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Technically, email *is* federated. \*giggles\*

File: src/Model/Contact.php [hunk 1]
```diff
@@ -1458,15 +1458,26 @@ public static function getIdForURL($url, $uid = 0, $no_update = false, $default
 		if (DBA::isResult($contact)) {
 			$contact_id = $contact["id"];
 
-			// Update the contact every 7 days
-			if (in_array($contact['network'], Protocol::NATIVE_SUPPORT)) {
+			// Update the contact every 7 days (Don't update mail or feed contacts)
+			if (in_array($contact['network'], Protocol::FEDERATED)) {
```
Comment (MrPetovan) [inline src/Model/Contact.php]: Technically, email *is* federated. \*giggles\*

**Comment by annando** [comment]
But you cannot use the regular probing mechanism.

**Comment by MrPetovan** [comment]
I know, it's past midnight, I had a bad day, I'm letting steam off by being facetious on GitHub. It's your Sunday morning, you can deal with it.

**Comment by annando** [comment]
No problem. I cross my fingers that tomorrow will be a better day for you.

BTW: I just woke in the middle of the night and started some work. Hopefully I now will find some more sleep before getting up.

**Comment by MrPetovan** [comment]
Thanks, now I feel like an ass for going hard on you. Sleep well, friend.

**Comment by annando** [comment]
Really no problem!

**Pull request merged**